### PR TITLE
GH-602 user resource deprecation

### PR DESCRIPTION
**Pull request by danielmkn**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,9 @@
+## 6.24.2 (January 12, 2023). Tested on Artifactory 7.49.5
+
+DEPRECATION:
+
+* resource/artifactory_user and resource/artifactory_unmanaged_user: added deprecation notice. Both resources are deprecated in favor of resource/artifactory__managed_user.
```
Comment (alexhung) [inline CHANGELOG.md:5]: There's an extra underscore in `artifactory_managed_user`.

**Review by alexhung — CHANGES_REQUESTED** [review]
Please work with Shailendra and determine the appropriate deprecation period (6 months?).

I'd also recommend creating a new discussion thread in https://github.com/jfrog/terraform-provider-artifactory/discussions and ask the users in previous [Password Issue](https://github.com/jfrog/terraform-provider-artifactory/discussions/333) discussion.

**Comment by danielmkn** [comment]
Discussion is here - #623

**Review by Turhsus — APPROVED** [review]
Other than the deprecation period, LGTM

**Comment by danielmkn** [comment]
There are a bunch of tests, using `artifactory_user` without the password provided. We will have to replace them before the merge if it will be decided to proceed.

**Comment by danielmkn** [comment]
We won't deprecate the resource, based on the conversation with our users. I'll update the documentation instead in a separate PR.

**Pull request closed without merging**